ext4: check ext4_journal_get_write_access() errors

From: Akinobu Mita <akinobu.mita@gmail.com>

Check ext4_journal_get_write_access() errors.

Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
Cc: Stephen Tweedie <sct@redhat.com>
Cc: adilger@clusterfs.com
Cc: Andrew Morton <akpm@linux-foundation.org>
Cc: Mingming Cao <cmm@us.ibm.com>
---
 fs/ext4/namei.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

Index: linux-2.6.25-rc9/fs/ext4/namei.c
===================================================================
--- linux-2.6.25-rc9.orig/fs/ext4/namei.c	2008-04-14 11:11:55.000000000 -0700
+++ linux-2.6.25-rc9/fs/ext4/namei.c	2008-04-14 11:12:24.000000000 -0700
@@ -57,10 +57,15 @@ static struct buffer_head *ext4_append(h
 
 	*block = inode->i_size >> inode->i_sb->s_blocksize_bits;
 
-	if ((bh = ext4_bread(handle, inode, *block, 1, err))) {
+	bh = ext4_bread(handle, inode, *block, 1, err);
+	if (bh) {
 		inode->i_size += inode->i_sb->s_blocksize;
 		EXT4_I(inode)->i_disksize = inode->i_size;
-		ext4_journal_get_write_access(handle,bh);
+		*err = ext4_journal_get_write_access(handle, bh);
+		if (*err) {
+			brelse(bh);
+			bh = NULL;
+		}
 	}
 	return bh;
 }
